autoload/neomake/utils.vim

```diff
@@ -200,3 +200,20 @@ function! neomake#utils#GetHighlight(group, what) abort
   endif
   return val
 endfunction
+
+function! neomake#utils#ExpandArgs(args) abort
+    if neomake#utils#IsRunningWindows()
+        call neomake#utils#ExpandArgsInWindows(a:args)
+    else
+        call neomake#utils#ExpandArgsInLinux(a:args)
+    endif
+endfunction
+
+function! neomake#utils#ExpandArgsInWindows(args) abort
+    " Don't expand &shellcmdflag argument of cmd.exe nor arguments like '"something"'
+    call map(a:args, "(v:val ==? &shellcmdflag || v:val =~? '.*\".*\".*') ? v:val : expand(v:val)")
+endfunction
+
+function! neomake#utils#ExpandArgsInLinux(args) abort
+    call map(a:args, 'expand(v:val)')
```

Comment:
FWIW, `args` can be a callback also - but I am not sure if it gets invoked in the right context.
But that is the way we should do it.

The current misuse of `expand` also prevents you from using something like `sh -c 'date +%N`..